### PR TITLE
User can customise mark

### PR DESCRIPTION
**Pull request by laurentqro**
screen shot 2018-05-16 at 12 19 11

### CODE REVIEW

File: src/main/java/TicTacToe/FakePlayer.java [hunk 1]
```diff
@@ -21,8 +22,22 @@ public String getInput() {
...
     }
+
+    @Override
+    public String getMarkChoice() {
+        return markChoices.pop();
+    }
+
+    void setMarkChoices(String... inputs) {
+        LinkedList<String> choices = new LinkedList<>(Arrays.asList(inputs));
+        this.markChoices = choices;
```
Comment (christophgockel) [inline src/main/java/TicTacToe/FakePlayer.java:37]: Not a biggie, but `choices` can be inlined as it seems.

File: src/main/java/TicTacToe/Player.java [hunk 2]
```diff
@@ -18,4 +18,14 @@ public String getMark() {
     public String getInput() {
         return display.getInput();
     }
+
+    @Override
+    public String getMarkChoice() {
```
Comment (christophgockel) [inline src/main/java/TicTacToe/Player.java:23]: Is it important to have separate `getInput()` and `getMarkChoice()` methods? In both instances we're asking the player to provide input, right?

File: src/main/java/TicTacToe/Session.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            markChoice = sanitize(player.getMarkChoice());
+            if(isValid(markChoice)) {
+                player.setMark(markChoice);
+                break;
+            }
+            display.warnInvalidInput(markChoice);
+        }
+    }
+
+    private boolean isValid(String input) {
```
Comment (christophgockel) [inline src/main/java/TicTacToe/Session.java]: Similar to our previous discussion it seems that verifying and sanitizing input in this instance is more of a `display` concern, no? The `session` requests a mark choice from the display. Ideally a valid one already.

File: src/main/java/TicTacToe/Session.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package TicTacToe;
+
+class Session {
```
Comment (christophgockel) [inline src/main/java/TicTacToe/Session.java]: What's the intent behind a `Session`? Right now it feels like its only responsibility is to change the player's symbols _after_ they were given default ones already.
I wonder if that's enough to necessitate a separate class?

**Comment by andreamazza89** [comment]
Your input validation logic is scattered around two different classes (Game and Session), which is fine given its scope is fairly contained, but it might make sense to move this into a class of its own.

Also, it could be interesting to think about where the responsibility should lie for what constitutes a valid move: at the moment the Game class "knows" that any of 9 strings ("1" - "9") are valid input; in a similar way, the Session knows that the maximum length for a symbol is 2-long.
I think that in both cases the knowledge of what is 'valid' input should maybe come from whatever it is that you "give" that input to.

Sorry if this is a bit cryptic and no need to change anything based on my comment; let's discuss outside of this PR if you want.

**Comment by laurentqro** [comment]
@andreamazza89, I think I understand.

The Board should determine what is a valid move (1-9)
The Cell should determine what mark it accepts and whether it can legally be marked

did I understand correctly?

**Comment by andreamazza89** [comment]
👍 I think so.

**Pull request merged**